File: Assets/Tests/KotlinTokenizer/properties.swift

protocol Hello {
    var foo: String { get }
    var bar: String { get set }
}

class A {
    var myVar: String?
    var stateObservable1: Observable<RestaurantsListState> { return state.asObservable() }
    var stateObservable2: Observable<RestaurantsListState> {
        get {
            return state.asObservable()
        }
    }
    var stateObservable3: Observable<RestaurantsListState> {
        NSLog("Multiple statements")
        return state.asObservable()
    }
    
    var center: Point {
        get {
            return Point(x: centerX, y: centerY)
        }
        set {
            origin.x = newValue.x - 100
        }
    }

    var top: Point {
        get {
            return Point(x: topX, y: topY)
        }
        set(val) {
            origin.y = 0
            origin.x = val.x
        }
    }

    var subject: TestSubject!

    lazy var players: [String] = {
        var temporaryPlayers = [String]()
        temporaryPlayers.append("John Doe")
        return temporaryPlayers
    }()
    
    private lazy var name: String = {() -> String in
        return "abc"
    }()
    
    var isLocating = false {
        didSet {
            delegate.set(isLocating: isLocating)
        }
    }

    private(set) var anotherName: String?

    private(set) var anotherNameWithDidSet: String = "a value" {
        didSet {}
    }

}

struct Rect {
    var origin = Point()
    var size = Size()
    
    var center: Point {
        get {
            let centerX = origin.x + (size.width / 2)
            let centerY = origin.y + (size.height / 2)
            return Point(x: centerX, y: centerY)
        }
        set(newCenter) {
            origin.x = newCenter.x - (size.width / 2)
            origin.y = newCenter.y - (size.height / 2)
        }
    }
}

class StepCounter {
    var totalSteps: Int = 0 {
        willSet(newTotalSteps) {
            print("About to set totalSteps to \(newTotalSteps)")
        }
        didSet {
            if totalSteps > oldValue  {
                print("Added \(totalSteps - oldValue) steps")
            }
        }
    }
}

File: Assets/Tests/KotlinTokenizer/properties.kt

internal interface Hello {
    val foo: String
    var bar: String
}

internal class A {
    internal var myVar: String? = null
    internal val stateObservable1: Observable<RestaurantsListState>
        get() = state.asObservable()
    internal val stateObservable2: Observable<RestaurantsListState>
        get() = state.asObservable()
    internal val stateObservable3: Observable<RestaurantsListState>
        get() = state.asObservable()
    internal val stateObservable4: Observable<RestaurantsListState>
        get() {
            NSLog("Multiple statements")
            return state.asObservable()
        }
    internal var center: Point
        get() = Point(x = centerX, y = centerY)
        set(newValue) {
            origin.x = newValue.x - 100
        }
    internal var top: Point
        get() = Point(x = topX, y = topY)        
        set(val) {
            origin.y = 0
            origin.x = val.x
        }
    lateinit internal var subject: TestSubject
    internal val players: List<String> by lazy {
        var temporaryPlayers = listOf<String>()
        temporaryPlayers.append("John Doe")
        temporaryPlayers
    }
    private val name: String by lazy {   -> 
        "abc"
    }
    internal var isLocating = false
        set(newValue) {
            val oldValue = field
            field = newValue
            delegate.set(isLocating = isLocating)
        }
    var anotherName: String? = null
        private set
    var anotherNameWithDidSet: String = "a value"
        private set(newValue) {
            field = newValue
        }
    internal val myVar: String
        get() {
            if (a == 5) {
                return "test"
            } else {
                return "b"
            }
        }
}

internal data class Rect(
    internal var origin = Point(),
    internal var size = Size()) {
    internal var center: Point
        get() {
            val centerX = origin.x + (size.width / 2)
            val centerY = origin.y + (size.height / 2)
            return Point(x = centerX, y = centerY)
        }
        set(newCenter) {
            origin.x = newCenter.x - (size.width / 2)
            origin.y = newCenter.y - (size.height / 2)
        }
}

internal class StepCounter {
    internal var totalSteps: Int = 0
        set(newTotalSteps) {
            val oldValue = field
            print("About to set totalSteps to ${newTotalSteps}")
            field = newTotalSteps
            if (totalSteps > oldValue) {
                print("Added ${totalSteps - oldValue} steps")
            }
        }
}
